src/model/loop.ml

```diff
@@ -500,9 +500,9 @@ module Make
           st, acc
         )
 
-  let check_solve st ~(file : _ Dolmen_loop.State.file) ~loc l =
+  let check_solve st ~(file : _ Dolmen_loop.State.file) ~loc (hyps, _) =
```

Comment:
It's wrong to ignore the goals here ! The goals should be evaluated to see if at least one of them is satisfied !